expression/builtin_json_vec.go

```diff
@@ -27,11 +29,36 @@ func (b *builtinJSONDepthSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
 }
 
 func (b *builtinJSONKeysSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinJSONKeysSig) vecEvalJSON(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETJson, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalJSON(b.ctx, input, buf); err != nil {
+		return err
+	}
+
+	result.ReserveJSON(n)
+	var j json.BinaryJSON
+	for i := 0; i < n; i++ {
+		if buf.IsNull(i) {
+			result.AppendNull()
+			continue
+		}
+
+		j = buf.GetJSON(i)
+		if j.TypeCode != json.TypeCodeObject {
+			result.AppendNull()
```

Comment:
Should we return a `json.ErrInvalidJSONData` error in this case?

Reply:
yes, I should.
But I don't know whether or not it should return the error immediately in the `for` loop, or return the error at the end of the function?

Reply:
I also have the same confusion in other vectorized function

Reply:
Normally, if ```err != nil```, we need to return error directly. But there are some cases where the error is reprocessed, it should recheck the ```err```.